ui/src/main/java/org/pentaho/di/ui/spoon/delegates/SpoonDBDelegate.java

```diff
@@ -525,6 +535,23 @@ public void newConnection( HasDatabasesInterface hasDatabasesInterface ) {
     }
   }
 
+  private void saveConnectionToRepository( DatabaseMeta databaseMeta ) {
```

Comment:
this should only be needed on Edit or Undo

Reply:
same as above